src/npctalk.cpp

```diff
@@ -5460,314 +5483,156 @@ talk_effect_t::talk_effect_t( const JsonObject &jo, const std::string &member_na
     }
 }
 
+static const
+std::vector<std::tuple<std::string, std::string, jarg, void ( talk_effect_fun_t::* )( const JsonObject &, const std::string &, bool )>>
+un_parsers = {
+    std::make_tuple( "u_add_effect", "npc_add_effect", jarg::member, &talk_effect_fun_t::set_add_effect ),
```

Comment:
Use modern syntax:
```suggestion
    { "u_add_effect", "npc_add_effect", jarg::member, &talk_effect_fun_t::set_add_effect },
```
And please use a real struct. Tuples and their `std::get<>` syntax are pretty ugly and there's no reason to prefer them here.

Also, for a container of this size, `std::map` is likely better (just like in the [math parser](https://github.com/CleverRaven/Cataclysm-DDA/blob/0b00ccbd3008bc0f7f6dabea02e26002ea3d258a/src/math_parser_diag.h#L123-L139)).

Reply:
At this point, the key string have not been determined, so I thought that using map would actually slow things down. Once we adopt the vector. Please let me know if there is a better way.